src/DIRAC/WorkloadManagementSystem/Executor/InputData.py

```diff
@@ -379,8 +385,6 @@ def __getSiteCandidates(self, okReplicas, vo):
                     if seStatus["DiskSE"]:
                         # Sets contain only unique elements, no need to check if it's there
                         diskLFNs.add(lfn)
-                        if lfn in tapeLFNs:
-                            tapeLFNs.remove(lfn)
                     if seStatus["TapeSE"]:
                         if lfn not in diskLFNs:
                             tapeLFNs.add(lfn)
```

Comment:
I think this is assuming an that disk SEs appear before tape SEs? Would it be better to remove the disk replicas from `sitesData[siteName]["disk"]` after the `for seName in replicas:` loop is finished?

Reply:
Do you mean to removed disk replicas from `sitesData[siteName]["tape"]`?

Still, I am not sure this is best...